### PR TITLE
feat: finer user agent

### PR DESCRIPTION
**Pull request by maelle**
Fix #119 (the rebasing would have been messy)
Related to #114

### CODE REVIEW

**Comment by maelle** [comment]
@salvafern could you please review this? Thank you!

**Comment by maelle** [comment]
@salvafern hmmm actually do we really want to make dplyr an Imports?

File: R/EMODnetWFS-package.R [hunk 1]
```diff
@@ -6,6 +6,20 @@
...
+
 emodnetwfs_user_agent <- function() {
-  "EMODnetWFS R package https://github.com/EMODnet/EMODnetWFS"
+  version <- as.character(packageVersion("EMODnetWFS"))
+
+  if (nzchar(Sys.getenv("EMODNETWFS_CI"))) {
+  	return(sprintf("EMODnetWFS R package %s CI https://github.com/EMODnet/EMODnetWFS", version))
+  }
+
+  if (whoami::gh_username() %in% emodnetwfs_collaborators()) {
```
Comment (salvafern) [inline R/EMODnetWFS-package.R:20]: I see an error when using this when I have not set one of GITHUB_USERNAME or GITHUB_PAT. I have, however, the email available. It stops at `lookup_gh_username(email, token)`

> Error in fallback_or_stop(fallback, "Cannot find GitHub username for email") : 
>   object 'fallback' not found

Fun: `whoami::gh_username()`
```r
function (token = NULL, fallback = NULL) 
{
  env_gh_username <- Sys.getenv("GITHUB_USERNAME")
  if (nzchar(env_gh_username)) 
    return(env_gh_username)
  email <- try(email_address(), silent = TRUE)
  if (ok(email)) {
    if (!grepl("@", email)) {
      return(fallback_or_stop(fallback, "This does not seem to be an email address"))
    }
    lookup_gh_username(email, token)
  }
  else {
    fallback_or_stop(fallback, "Cannot get GitHub username")
  }
}
```

@maelle is this something to be reported or am I using it wrong?

**Review by salvafern — COMMENTED** [review]
All good for me except this comment: https://github.com/EMODnet/EMODnetWFS/pull/125/files#r1093047711

@maelle Any ideas? :)

File: R/EMODnetWFS-package.R (hunk 1 above)
Comment (maelle) [reply R/EMODnetWFS-package.R:20]: what does `whoami::email_address()` return?

Comment (maelle) [reply R/EMODnetWFS-package.R]: in any case it was bad that I hadn't added a fallback, now done.

Comment (salvafern) [reply R/EMODnetWFS-package.R]: returns my VLIZ email

**Review by salvafern — CHANGES_REQUESTED** [review]
keeps failing, I can see it is at `lookup_gh_username`: https://github.com/r-lib/whoami/blob/6783679bc4d0924f267dc18f4e9be3222b37a918/R/whoami.R#L321

```r
debug(emodnetwfs_user_agent)
debug(whoami::gh_username)
lookup_gh_username("some@email.com", NULL)
# Error in fallback_or_stop(fallback, "Cannot find GitHub username for email") : 
# 	object 'fallback' not found
```

**Review by salvafern — APPROVED** [review]
The exception works great now with `try()`. Thanks!!

**Pull request merged**